native-sdk/src/resource/vault.rs

```diff
@@ -7,13 +7,90 @@ use radix_engine_interface::types::*;
 use sbor::rust::collections::BTreeSet;
 use sbor::rust::fmt::Debug;
 
-pub struct Vault(pub Own);
+// TODO: split impl
 
-impl Vault {
-    pub fn sys_new<Y, E: Debug + ScryptoDecode>(
+pub trait NativeVault {
+    fn create<Y, E: Debug + ScryptoDecode>(
         resource_address: ResourceAddress,
         api: &mut Y,
-    ) -> Result<Self, E>
+    ) -> Result<Vault, E>
+    where
+        Y: ClientApi<E>;
```

Comment:
Incidental comment - not suggesting we change this as part of this PR!

For all of these, I'd be tempted to consider ditching the `Y` generic and using `api: &mut impl ClientApi<E>` which would save a few lines on each definition :). Also - I'm really not sure why ClientApi is generic over the error type? It appears E is always RuntimeError, so we could remove that too.

Perhaps if we wanted to change this we could fix them all together - I'm not suggesting it's a big priority though.

Reply:
It's because we use it for both `native-sdk` and `scrypto`. They produce different errors.

Reply:
Got it - although in `native-sdk`, I think all these generics are quite distracting, and would suggest it would be cleaner if we could either have a concrete type for the whole ClientApi, or a trait "alias" `NativeClientApi` which hardcodes `E = RuntimeError` (similar to eg the ScryptoEncode trait)